eureka2-core/src/main/java/com/netflix/eureka2/transport/base/BaseMessageConnection.java

```diff
@@ -174,6 +177,11 @@ public Boolean call(Object message) {
             public void call(Object o) {
                 metrics.incrementIncomingMessageCounter(o.getClass(), 1);
             }
+        }).doOnTerminate(new Action0() {
+            @Override
+            public void call() {
+                shutdown();
```

Comment:
Do we have verification tests to ensure that onComplete and onError terminates will trigger the proper shutdown and send the correct onComplete or onError on the lifecycle observable?